lib/Common/PlatformAgnostic/AssemblyCommon.h

```diff
@@ -0,0 +1,30 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+#ifndef _WIN32
+
+extern "C" void __register_frame(const void* ehframe);
+extern "C" void __deregister_frame(const void* ehframe);
+
+#if defined(_AMD64_) && !defined(DISABLE_JIT)
```

Comment:
Just noticed this part. ENABLE_NATIVE_CODEGEN is a subset of DISABLE_JIT. ENABLE_NATIVE_CODEGEN is generally meant for JIT-related code directly. DISABLE_JIT was added as part of doing the Interpreter-only version of ChakraCore last year- it implies !ENABLE_NATIVE_CODEGEN but also disables other things like dynamic interpreter thunks, interpreter profiling etc.